### PR TITLE
KBS protocol: Add function to get attestation token

### PR DESCRIPTION
**Pull request by jialez0**

### CODE REVIEW

File: kbs_protocol/src/lib.rs [hunk 1]
```diff
@@ -177,6 +181,18 @@ impl KbsRequest for KbsProtocolWrapper {
 
         bail!("Request KBS: Attested but KBS still return Unauthorized")
     }
+
+    async fn attest(&mut self, url_string: String) -> Result<String> {
+        let url = Url::parse(&url_string)
```
Comment (jepio) [inline kbs_protocol/src/lib.rs:186]: why do we need to parse the url and not pass it directly to `self.attestation`? I don't know what you imagine the caller of this function would do.

**Review by jepio — COMMENTED** [review]
This is simple enough, but I don't know whether this is useful just by looking at the interface. Where do you want to call this from?

**Comment by jialez0** [comment]
@jepio We are gradually increasing the ability to use passport models for the attestation system, this API is designed to serve passport model scenarios. When the caller needs to obtain an attestation token, they can directly call this API without worrying about the complex evidence acquisition process and HTTPS interaction process.

**Comment by jialez0** [comment]
> This is simple enough, but I don't know whether this is useful just by looking at the interface. Where do you want to call this from?

Now, the client tool of KBS is using this interface: https://github.com/confidential-containers/kbs/blob/main/tools/client/src/main.rs#L99, and in the near future, AA will also provide a public API for upper level applications to obtain attestation token.

File: kbs_protocol/src/lib.rs [hunk 2]
```diff
@@ -21,7 +21,10 @@ pub const KBS_URL_PREFIX: &str = "kbs/v0";
 
 #[async_trait]
 pub trait KbsRequest {
-    async fn http_get(&mut self, url: String) -> Result<Vec<u8>>;
+    /// Get confidential resource
+    async fn http_get(&mut self, resource_url: String) -> Result<Vec<u8>>;
+    /// Attestation and get attestation results token (Base64 endcoded)
```
Comment (wainersm) [inline kbs_protocol/src/lib.rs:26]: s/endcoded/encoded/

File: kbs_protocol/src/lib.rs [hunk 3]
```diff
@@ -113,7 +116,8 @@ impl KbsProtocolWrapper {
         match attest_response.status() {
             reqwest::StatusCode::OK => {
                 self.authenticated = true;
-                Ok(())
+                let token = attest_response.json::<serde_json::Value>().await?["token"].to_string();
```
Comment (wainersm) [inline kbs_protocol/src/lib.rs:119]: Would make sense to check here the token is really base64 encoded?

Comment (jepio) [reply kbs_protocol/src/lib.rs:119]: the token is supposed to be a json-web-token which has base64 components but is simply a base64 encoded string.

**Comment by mythi** [comment]
KBS's `tools/client` depends on this so it'd be nice to have this in the release so that `tools/client` can be updated.

**Comment by jialez0** [comment]
@sameo Could you PTAL?

**Review by jepio — APPROVED** [review]

**Review by sameo — APPROVED** [review]

**Pull request merged**